exonum/src/storage/keys.rs

```diff
@@ -388,4 +410,86 @@ mod tests {
         // Notice the different order of values compared to the previous test
         assert_eq!(index.values().collect::<Vec<_>>(), vec![100, 200]);
     }
+
+    #[test]
+    fn test_storage_key_for_system_time_round_trip() {
+        use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+        let times = [
+            UNIX_EPOCH,
+            UNIX_EPOCH + Duration::new(13, 23),
+            SystemTime::now(),
+            SystemTime::now() + Duration::new(17, 15),
+        ];
+
+        let mut buffer = [0u8; 12];
+        for time in times.iter() {
+            time.write(&mut buffer);
+            assert_eq!(*time, SystemTime::read(&buffer));
+        }
+    }
+
+    #[test]
+    fn test_storage_key_for_system_time_ordering() {
+        use std::time::{Duration, SystemTime};
+
+        let (mut buffer1, mut buffer2) = ([0u8; 12], [0u8; 12]);
+        for _ in 0..FUZZ_SAMPLES {
+            let time1 = SystemTime::now();
+            let time2 = time1 + Duration::new(0, 1);
+            time1.write(&mut buffer1);
+            time2.write(&mut buffer2);
+            assert!(buffer1 < buffer2);
+        }
+    }
```

Comment:
I am not sure, but I think that this test will *not* fail, if we reverse the order in which we write seconds and nanos. If this is true, that it's better to add a test that can detect such behavior. 

For example, generating `FUZZ_SAMPLE`s *random* times and than checking that the ordering is correct should work I guess?

Reply:
:+1: I'd say "fuzz" implies randomness involved.

Reply:
> I'd say "fuzz" implies randomness involved.

Generating radom samples and than adding one nanosecond to them would not help to check that seconds and nanons are compared in the correct order. It's important to compare two random value.